tests/test_attribute_definition.py

```diff
@@ -150,11 +138,8 @@ def print_parameter(self, printer: Printer) -> None:
         printer.print_string("]")
 
     def verify(self) -> None:
-        if not isinstance(self.data, list):
-            raise VerifyException("int_list data should hold a list.")
-        for elem in self.data:
-            if not isinstance(elem, int):
-                raise VerifyException("int_list list elements should be integers.")
+        # We must override verify on Attribute
```

Comment:
@math-fehr  is there a reason this must be done? Should I file an issue to remove this requirement?

Reply:
What error do you get? We shouldn't have to keep it.

Reply:
```
_________________________________________________________________________________________________ ERROR collecting tests/test_attribute_definition.py __________________________________________________________________________________________________
tests/test_attribute_definition.py:123: in <module>
    @irdl_attr_definition
xdsl/irdl.py:1812: in irdl_attr_definition
    return irdl_data_definition(cls)  # type: ignore
xdsl/irdl.py:1639: in irdl_data_definition
    raise Exception(
E   Exception: In IntListData definition: Cannot infer "verify" method. Type parameter of Data is not a class.
=============================================================================================================== short test summary info ================================================================================================================
ERROR tests/test_attribute_definition.py - Exception: In IntListData definition: Cannot infer "verify" method. Type parameter of Data is not a class.
```

Reply:
Oh I see! That's some code I wrote years ago!
In `Data` it seems we are checking during construction that the type is the expected one.
Since we now trust our types, maybe we should remove that? Let's do that in another PR though.

Reply:
https://github.com/xdslproject/xdsl/issues/1075